org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java

```diff
@@ -921,6 +930,87 @@ public void postVisit(ASTNode node) {
 		return result;
 	}
 
+	private List<String> replaceSafeSystemOption(List<String> options) {
+		int ind = -1;
+		String[] arr = options.toArray(new String[options.size()]);
+		for( int i = 0; i < options.size(); i++ ) {
+			if(options.get(i).equals("--system")) {
+				ind = i + 1;
+			}
+			arr[i] = options.get(i);
+		}
+		if( ind == -1 ) {
+			return options;
+		}
+		String existingVal = arr[ind];
+		if( Paths.get(existingVal).toFile().isDirectory()) {
+			return options;
+		}
+
+//		// Replace the index with a new safer one
+		Bundle bundle = Platform.getBundle("org.eclipse.jdt.core.javac");
+		IPath stateLocation = Platform.getStateLocation(bundle);
+		IPath systemLib = stateLocation.append("systemlib");
+		IPath safePath = systemLib.append("dummy");
+		IPath lib = safePath.append("lib");
+		IPath jrt = lib.append("jrt-fs.jar");
+		IPath modules = lib.append("modules");
+
+		if( !systemLib.toFile().exists()) {
+			systemLib.toFile().mkdirs();
+		}
+		if( !safePath.toFile().exists()) {
+			safePath.toFile().mkdirs();
+		}
+		if( !lib.toFile().exists()) {
+			lib.toFile().mkdirs();
+		}
+
+
+		try {
+			if( !jrt.toFile().exists())
+				Files.createFile(jrt.toFile().toPath());
+			if( !modules.toFile().exists())
+				Files.createFile(modules.toFile().toPath());
+        } catch (FileAlreadyExistsException e) {
+        	// Ignore
+        } catch (IOException e) {
+        	// Ignore
+		}
+		arr[ind] = safePath.toString();
+		return Arrays.asList(arr);
+	}
+
+	public static String toSafeFileName(String path) {
```

Comment:
Can you please describe as a comment in the code the actual issue in OpenJDK which now require this, ideally pointing to the bit of code that is broken in case of special characters?

Reply:
method is removed; was left over from previous version where I thought I actually needed to use the jars in some way.